### DISCUSSION TITLE
FC gradient  increases a lot during a minimization

### DISCUSSION THREAD

**Discussion by Lanlin123** [SSCHAcode/python-sscha#109] [Q&A] [answered]
Dear developers,
    In my free energy hessian calculation, the FC gradient always increased a lot during the minimization. I have tried the methods from the FAQS part of your sscha website, i.e. , increasing the number of configuration to 2000 and increasing the kong-liu ratio, but it didn't work. I wonder if I must do the vc-relax calcution before calculating free energy hessian. Would this matter to  the FC gradient increasing problem. Below is part of my minimization result. 

![image](https://user-images.githubusercontent.com/51743630/193235101-6f9092dc-a5f9-4f19-9158-4f25c3372dda.png)
![image](https://user-images.githubusercontent.com/51743630/193235216-2b4d3e14-71a5-4c69-9b34-af7722cdd12a.png)
![image](https://user-images.githubusercontent.com/51743630/193235288-4d287972-5cff-4e03-864d-b78e50e2c1d2.png)
![image](https://user-images.githubusercontent.com/51743630/193235331-0e23ae15-43fa-472b-b926-afa818ed7643.png)
![image](https://user-images.githubusercontent.com/51743630/193235497-4e02ce54-5a3b-40ed-958d-567b159677c3.png)

waiting for your reply.
Best regards,
Lanlin.

**Comment by mesonepigreco** [comment, marked as answer]
Hi, In this case, the minimization seems to be working correctly as the free energy is decreasing.
I suggest continuing the minimization with new populations (probably, you can even reduce the number of configurations to speed up). 

If you have a strongly anharmonic system, it is possible that the gradient increases in the first few populations. 
The calculation of the free energy hessian should be done after the free energy is minimized. 

Bests,
Lorenzo

**Reply by Lanlin123** [reply]
Thank you, Lorenzo. So before the calculation of free energy hessian, vc-relax using sscha is not necessary,right? I worry the cell without vc-relax will always give negative free energy hessian like the case in phonon caculation using DFPT.

Bests,
Lanlin.

**Reply by mesonepigreco** [reply]
Hi, it depends on the system, but it is always better to relax completely the free energy (even with vc-relax) before computing the free energy hessian. However, usually the free energy hessian is less dependent on the cell relaxation with the SSCHA than DFPT phonons if the initial cell was already relaxed without quantum/thermal effects.
